subway_structure.py

```diff
@@ -1738,9 +1740,9 @@ def __iter__(self):
 
     @property
     def is_good(self):
-        assert self.validate_called, (
+        assert self.errors or self.validate_called, (
```

Comment:
Better to change to 
```python
is self.errors or self.validate_called:
    raise Exception(
        "You mustn't refer to City.is_good property before calling "
        "the City.validate() method unless an error already occurred."
    )
```
Otherwise, it will not throw an exception, if self.validate_called = False and self.errors is not empty. We want to get an exception, if self.errors is not empty, right?

Reply:
Both, because:
1. Asserts should be used only in tests/debug: https://pycharm-security.readthedocs.io/en/latest/checks/AST100.html
2. AssertionError is too wide, I guess. We can use it for many things, not only to check our code organization. While RuntimeError says that there's something wrong with program's logic. And that is our case.

Reply:
Done.